.github/workflows/push-docker.yaml

```diff
@@ -61,3 +69,84 @@ jobs:
         if: ${{ ! inputs.dry-run }}
         run: |
           docker push --all-tags "${{ inputs.repository-name }}"
+  push-docker-multi-arch:
+    name: Push Semgrep Docker Image (multi-arch)
+    runs-on: ubuntu-22.04
+    if: ${{ inputs.multi-arch }}
+    steps:
+      - uses: iarekylew00t/regctl-installer@v1
+      - uses: actions/download-artifact@v3
+        with:
+          path: /tmp/artifacts
+      - name: Merge images
+        env:
+          MERGED_IMAGE_PATH: /tmp/merged-image
+        run: |
+          mkdir -p ${MERGED_IMAGE_PATH}/blobs/sha256
+
+          # Find docker image artifacts
+          image_filenames=(/tmp/artifacts/${{ inputs.artifact-name }}-arch-*/image.tar)
+          if [[ ${#image_filenames[@]} == 0 ]]; then
+            echo "No artifacts found matching ${{ inputs.artifact-name }}-arch-*"
+            exit 1
+          else
+            echo "Found images: ${image_filenames[@]}"
+          fi
+
+          # Create an empty manifest list. This will ultimately be the union of all the arch-specific image manifests we intend to publish.
```

Comment:
A link to some doc about multi-arch docker image would be useful. I have no idea what are those manifest file
for example, and whoever will have to modify this file in 6 months could use some help to accelerate the learning.